nntrainer/layers/preprocess_translate_layer.h

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file	preprocess_layer.h
+ * @date	31 December 2020
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is Preprocess Translate Layer Class for Neural Network
+ *
+ */
+
+#ifndef __PREPROCESS_TRANSLATE_LAYER_H__
+#define __PREPROCESS_TRANSLATE_LAYER_H__
+#ifdef __cplusplus
+
+#include <random>
+
+#if defined(ENABLE_DATA_AUGMENTATION_OPENCV)
+#include <opencv2/highgui/highgui.hpp>
```

Comment:
instead of including header, how about declare `class cv::Mat;` here?

Reply:
That would allow compilation but would break while running. Wouldn't that be opposite of declaring dependency at compile-time?

Reply:
Correct me If I am wrong, those `cv::Mat` is private and used only inside `.cpp`. So there seems declaring here shouldn't cause any undefined reference error. if someone would access those variables, would have a runtime error but it seems the case would never happen.

Reply:
The `cv::Mat` object in private section creates objects in header, so forward declare wont work. 
There is a way to convert them to unique_ptr, but its too much for now as opencv dependency wont be kept for long anyway.

Reply:
Yes. That's too much. Thanks for the confirmation!